### DISCUSSION TITLE
timestamp for mySQL database

### DISCUSSION THREAD

**Discussion by aeroptl** [Budibase/budibase#13101] [Help] [answered]
Hello, 

I'm trying to insert a timestamp in mySQL query on budibase with the value of the binding called time being: 

{{ date now  "YYYY/MM/DD HH:mm:ss" }}

 I can't seem to get it too work. 

Does anyone know how I can insert a timestamp into a database using bindings/queries? without using date picker on forms. 

 Thanks.

**Comment by ConorWebb96** [comment]
Hey @aeroptl,

I was able to update a timestamp using the below query.

```sql
UPDATE timezone
SET created_at = {{timestampers}}
WHERE id = {{id}};
```

<img width="1259" alt="Screenshot 2024-02-26 at 12 05 40" src="https://github.com/Budibase/budibase/assets/126772285/b421e213-6e21-4e83-8e5d-38b21896a67f">

There isn't any helpers available in the query area. So you would need to format your data correctly before passing it back to the query.  Make sure whatever you are outputting fits the format used for date pickers. The formats of the dates used normally look like this; `2024-01-10T12:21:23.913Z`

I hope this helps.

**Reply by aeroptl** [reply]
Hi, Thanks for that

I understand the SQL query, it's the binding you've created called timestampers. Did you manually type that in? Because I'm looking for a function which can do that for me. 

I hope that makes sense.

**Reply by ConorWebb96** [reply, marked as answer]
Hey @aeroptl,

If you are passing it in from the frontend you can convert the time however you wish. Below is an example of how i've done this. 

- I've created a table component which outputs various info.
- I've added a button component thats attached to each row
- I've added 2 different actions an execute query action which uses the updated time query
- This has the `id` binding and the `timestampers` binding which I am dynamically passing in
- Id is simply getting the row id
- For timestampers I'm using the below JS script to pass in the current now time in the correct format.
```js
const currentDate = new Date();
const timestamp = currentDate.toISOString();

return timestamp;
```
<img width="1511" alt="Screenshot 2024-02-26 at 14 58 34" src="https://github.com/Budibase/budibase/assets/126772285/7b25150c-dd79-4cbd-94c0-57a7f9b304b2">
- Lastly I've added a refresh dataprovider action so you can see the data changing in real time.

**Demo of this working**
![20240226_145340](https://github.com/Budibase/budibase/assets/126772285/7751e70b-fd9e-4a75-a349-13a4a15f8b04)

I hope this helps.
